servatrice/src/serversocketinterface.cpp

```diff
@@ -971,6 +972,39 @@ Response::ResponseCode AbstractServerSocketInterface::cmdBanFromServer(const Com
     return Response::RespOk;
 }
 
+QString AbstractServerSocketInterface::parseEmailAddress(const std::string &stdEmailAddress)
+{
+    QString emailAddress = QString::fromStdString(stdEmailAddress);
+
+    const QRegularExpression rx(R"(\b([A-Za-z0-9._%+-]+)@([A-Za-z0-9.-]+\.[A-Za-z]{2,4})\b)");
+    const QRegularExpressionMatch match = rx.match(emailAddress);
+
+    if (emailAddress.isEmpty() || !match.hasMatch()) {
+        return QString();
```

Comment:
this obscures why their email was invalid from the user, if they enter an email that didn't match the regex it'll tell them they forgot to enter an email with no context